google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/deprecated/spi/PubSubRpc.java

```diff
@@ -40,189 +41,149 @@
 import com.google.pubsub.v1.PullResponse;
 import com.google.pubsub.v1.Subscription;
 import com.google.pubsub.v1.Topic;
-import java.util.concurrent.Future;
 
 public interface PubSubRpc extends AutoCloseable {
 
   /**
-   * A callback that can be registered to {@link PullFuture} objects. Objects of this class allow
-   * to asynchronously react to the success or failure of a pull RPC.
-   */
-  interface PullCallback {
-
-    /**
-     * This method is invoked with the result of a {@link PullFuture} when it was successful.
-     *
-     * @param response the pull response
-     */
-    void success(PullResponse response);
-
-    /**
-     * This method is invoked when the {@link PullFuture} failed or was cancelled.
-     *
-     * @param error the execption that caused the {@link PullFuture} to fail
-     */
-    void failure(Throwable error);
-  }
-
-  /**
-   * A {@link Future} implementation for pull RPCs. This class also allows users to register
-   * callbacks via {@link #addCallback(PullCallback)}.
-   */
-  interface PullFuture extends Future<PullResponse> {
```

Comment:
This is an "internal" component of Pub/Sub - it's not exposed on the `PubSub` interface, so I'm pretty sure users aren't expected to interact with it.